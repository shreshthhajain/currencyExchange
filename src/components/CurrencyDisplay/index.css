.inputStyle::placeholder {
    color: lightgray;
}

.invalidBalance {
    color:red;
}

.inputStyle {
    border: none;
    background: inherit;
    text-align: end;
}

.inputStyle:hover, .inputStyle:focus, .selectStyle:focus, .selectStyle:hover {
    border: none;
    outline: none;
}

input::-webkit-outer-spin-button,
input::-webkit-inner-spin-button {
    -webkit-appearance: none;
    margin: 0;
}

input[type=number] {
    -moz-appearance:textfield; /* Firefox */
}

.selectStyle {
    background: inherit;
    border: none;
    font-weight: bold;
    margin-left: -4px;
}

.outerDiv { 
    padding: 20px 10px 20px 10px;
    height: 70px;
    line-height: 30px;
}