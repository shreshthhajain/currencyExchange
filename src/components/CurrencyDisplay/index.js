import React from 'react';
import {FormattedNumber, IntlProvider} from 'react-intl';
import './index.css';

const CurrencyDisplay = (props) => {
    const backgroundColor = props.prefix === '-' ? '#fff' : '#ECECEC';
    return (
        <div className="outerDiv" style={{backgroundColor: backgroundColor}}>
        <div style={{display:'flex', justifyContent:'space-between'}}>
          <select value={props.currency} className="selectStyle" onChange={(event) =>props.onChangeCurrency(event)}>
            <option value="EUR">EUR</option>
            <option value="USD">USD</option>
            <option value="GBP">GBP</option>
          </select>
          <input type="number" className="inputStyle" value={props.amount} placeholder='0' onChange={(event) =>props.amountChange(event)} readOnly={props.readOnly}></input>
        </div>
        <div>Balance: <span className={props.prefix === '-' && props.amount > props.balance ? 'invalidBalance' : ''}><IntlProvider locale='en'><FormattedNumber style="currency" value={props.balance} currency={props.currency}/></IntlProvider></span> </div>
      </div>
    );
}
export default CurrencyDisplay;