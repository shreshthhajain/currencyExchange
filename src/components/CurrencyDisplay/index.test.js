import React from 'react';
import { shallow } from 'enzyme';
import jest from 'jest-mock';
import CurrencyDisplay from './index';
describe('CurrencyDisplay', () => {
  let component;
  beforeEach(() => {
    const props = {
        currency: 'EUR',
        onChangeCurrency: jest.fn(),
        balance: 100.00,
        amount: 1,
        amountChange: jest.fn(),
        readOnly: false,
        prefix: '-',
    }
    component = shallow(<CurrencyDisplay {...props} />);
  });
  it('should render correctly', () => {
    expect(component).toMatchSnapshot();
  });
});