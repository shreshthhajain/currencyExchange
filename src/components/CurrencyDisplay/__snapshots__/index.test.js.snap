// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrencyDisplay should render correctly 1`] = `
<div
  className="outerDiv"
  style={
    Object {
      "backgroundColor": "#fff",
    }
  }
>
  <div
    style={
      Object {
        "display": "flex",
        "justifyContent": "space-between",
      }
    }
  >
    <select
      className="selectStyle"
      onChange={[Function]}
      value="EUR"
    >
      <option
        value="EUR"
      >
        EUR
      </option>
      <option
        value="USD"
      >
        USD
      </option>
      <option
        value="GBP"
      >
        GBP
      </option>
    </select>
    <input
      className="inputStyle"
      onChange={[Function]}
      placeholder="0"
      readOnly={false}
      type="number"
      value={1}
    />
  </div>
  <div>
    Balance: 
    <span
      className=""
    >
      <IntlProvider
        defaultFormats={Object {}}
        defaultLocale="en"
        formats={Object {}}
        locale="en"
        messages={Object {}}
        onError={[Function]}
        textComponent={Symbol(react.fragment)}
      >
        <FormattedNumber
          currency="EUR"
          style="currency"
          value={100}
        />
      </IntlProvider>
    </span>
     
  </div>
</div>
`;
